src/Razor/Microsoft.NET.Sdk.Razor/src/ComputeCssScope.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Numerics;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class ComputeCssScope : Task
+    {
+        [Required]
+        public ITaskItem[] ScopedCssInput { get; set; }
+
+        [Required]
+        public string TargetName { get; set; }
+
+        [Output]
+        public ITaskItem[] ScopedCss { get; set; }
+
+        public override bool Execute()
+        {
+            ScopedCss = new ITaskItem[ScopedCssInput.Length];
+
+            for (var i = 0; i < ScopedCssInput.Length; i++)
+            {
+                var input = ScopedCssInput[i];
+                // Todo: Normalize path to forward slashes and lowercase before computing the hash
+                var relativePath = input.ItemSpec.Replace("\\","//");
+                var scope = input.GetMetadata("CssScope");
+                scope = !string.IsNullOrEmpty(scope) ? scope : GenerateScope(TargetName, relativePath);
+
+                var outputItem = new TaskItem(input);
+                outputItem.SetMetadata("CssScope", scope);
+                ScopedCss[i] = outputItem;
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        private string GenerateScope(string targetName, string relativePath)
+        {
+            using var hash = SHA256.Create();
+            var bytes = Encoding.UTF8.GetBytes(relativePath + targetName);
+            var hashBytes = hash.ComputeHash(bytes);
+
+            var builder = new StringBuilder();
+            builder.Append("b-");
+
+            builder.Append(ToBase36(hashBytes));
+
+            return builder.ToString();
+        }
+
+        private  string ToBase36(byte[] hash)
+        {
+            var builder = new StringBuilder();
+            const string chars = "abcdefghijklmnopqrstuvwxyz0123456789";
+            var dividend = new BigInteger(hash.AsSpan().Slice(0,8).ToArray());
+            while (dividend > 36)
+            {
+                dividend = BigInteger.DivRem(dividend, 36, out var remainder);
+                builder.Insert(0, chars[Math.Abs(((int)remainder))]);
+            }
+            return builder.ToString();
+        }
```

Comment:
This logic isn't right. The main issue is L63 - at minimum we should change this to `while (dividend > 0)` because otherwise the method fails to append the least-significant base-36 digit. For example, any input whose initial `dividend` value was in 0..36 would all produce the same empty-string output.

It would be better still if it was to pad the output to be a fixed length. It's strange that if the input is an array of zeros, then the output is an empty string. It would be better if it was the base36 equivalent of `00000000`.

I know in practice it's statistically unlikely that anyone will have immediate problems with the logic as-is, so I don't mind if we leave it unchanged for preview 8. However post-preview 8, can we change it to something like the following? It fixes both problems, and simplifies a couple of things:

```cs
private static string ToBase36(byte[] hash)
{
    const string chars = "0123456789abcdefghijklmnopqrstuvwxyz";

    var result = new char[10];
    var dividend = new BigInteger(hash.AsSpan().Slice(0, 8));
    for (var i = 0; i < 10; i++)
    {
        dividend = BigInteger.DivRem(dividend, 36, out var remainder);
        result[i] = chars[(int)remainder];
    }

    return new string(result);
}
```

Reply:
Yep, I'm collecting a list of follow-ups

Reply:
I didn't realize that you were using the span overload. I tried this and that doesn't exist in netstandard2.0 or something which we need for VS

Reply:
I've reverted the change to the original code and will address post preview8